src/main/scala/io/iohk/ethereum/vm/OpCode.scala

```diff
@@ -163,8 +163,11 @@ object OpCodes {
 }
 
 object OpCode {
-  def sliceBytes(bytes: ByteString, offset: Int, size: Int): ByteString =
-    bytes.slice(offset, offset + size).padTo(size, 0.toByte)
+  def sliceBytes(bytes: ByteString, offset: UInt256, size: UInt256): ByteString = {
+    val start = offset.min(bytes.size).toInt
+    val end = (offset + size).min(bytes.size).toInt
+    bytes.slice(start, end).padTo(size.toInt, 0.toByte)
```

Comment:
I think `.padTo(end - start, 0.toByte)` would make more sense. But this is what [geth does](https://github.com/ethereum/go-ethereum/blob/master/core/vm/common.go#L42). See the consequence in the tests.